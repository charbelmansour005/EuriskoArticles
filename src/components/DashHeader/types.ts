export type DashHeaderSearchProps = {
  search: string
  setSearch: (arg?: any) => void
  language: any
}
