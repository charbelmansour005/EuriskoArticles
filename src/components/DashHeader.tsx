import React from 'react';
import {View, StyleSheet} from 'react-native';
import DashSearchBar from './DashSearchBar';
import {themeColors} from '../helpers/themeColors';

type SearchProps = {
  search: string;
  setSearch: (arg?: any) => void;
};

const DashHeader = ({...props}: SearchProps): JSX.Element => {
  return (
    <View style={styles.headerContainer}>
      <View style={styles.headerTitleContainer}></View>
      <DashSearchBar
        search={props.search}
        onChangeText={val => props.setSearch(val)}
        onPressClear={() => props.setSearch('')}
      />
    </View>
  );
};

export default DashHeader;

const styles = StyleSheet.create({
  headerContainer: {
    backgroundColor: themeColors.orange,
    paddingHorizontal: 0,
    paddingBottom: 0,
  },
  headerTitleContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingTop: 0,
  },
});
