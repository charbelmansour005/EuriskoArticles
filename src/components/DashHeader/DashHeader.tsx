import React from 'react';
import {View, StyleSheet} from 'react-native';
import DashSearchBar from '../DashSearchBar/DashSearchBar';
import {themeColors} from '../../helpers/themeColors';
import {SearchProps} from './types';

const DashHeader = ({...props}: SearchProps): JSX.Element => {
  return (
    <View style={styles.headerContainer}>
      <View style={styles.headerTitleContainer}></View>
      <DashSearchBar
        search={props.search}
        onChangeText={val => props.setSearch(val)}
        onPressClear={() => props.setSearch('')}
      />
    </View>
  );
};

export default DashHeader;

const styles = StyleSheet.create({
  headerContainer: {
    backgroundColor: themeColors.white,
    paddingHorizontal: 0,
    paddingBottom: 0,
  },
  headerTitleContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingTop: 0,
  },
});
