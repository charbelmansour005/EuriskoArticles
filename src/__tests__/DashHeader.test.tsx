import React from 'react'
import renderer from 'react-test-renderer'
import DashHeader from '../components/DashHeader/DashHeader'

describe('DashHeader Component', () => {
  test('Component should show search bar', async () => {
    const tree = renderer
      .create(
        <DashHeader
          search={''}
          setSearch={function (arg?: any): void {
            throw new Error('Function not implemented.')
          }}
          language={undefined}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
