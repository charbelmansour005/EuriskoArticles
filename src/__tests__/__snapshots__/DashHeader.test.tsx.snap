// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should show search bar 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "paddingBottom": 0,
      "paddingHorizontal": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingTop": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderRadius": 0,
        "flexDirection": "row",
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 0,
        "width": "95%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/srh.png",
        }
      }
      style={
        Object {
          "height": 16,
          "marginLeft": 5,
          "marginRight": 5,
          "width": 16,
        }
      }
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Chercher des Articles"
      placeholderTextColor="gray"
      style={
        Object {
          "backgroundColor": "white",
          "color": "#0a0a0f",
          "width": "80%",
        }
      }
      value=""
    />
  </View>
</View>
`;

exports[`DashHeader Component Component should show search bar 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "paddingBottom": 0,
      "paddingHorizontal": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingTop": 0,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderRadius": 0,
        "flexDirection": "row",
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 0,
        "width": "95%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/srh.png",
        }
      }
      style={
        Object {
          "height": 16,
          "marginLeft": 5,
          "marginRight": 5,
          "width": 16,
        }
      }
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Chercher des Articles"
      placeholderTextColor="gray"
      style={
        Object {
          "backgroundColor": "white",
          "color": "#0a0a0f",
          "width": "80%",
        }
      }
      value=""
    />
  </View>
</View>
`;
